File: app/src/main/java/com/example/product_sale_app/data/ChatApiService.java
package com.example.product_sale_app.data;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface ChatApiService {
    @GET("api/chat")
    Call<BaseResponseModel<ChatMessageDto>> getMessages(
            @Query("pageIndex") int pageIndex,
            @Query("pageSize") int pageSize,
            @Query("chatBoxId") Integer chatBoxId,
            @Query("userId") Integer userId
    );

    @GET("api/chat/box/{chatBoxId}")
    Call<BaseResponseModel<ChatMessageDto>> getBoxMessages(
            @Path("chatBoxId") int boxId,
            @Query("pageIndex") int pageIndex,
            @Query("pageSize") int pageSize
    );

    @POST("api/chat")
    Call<BaseResponseModel<ChatMessageDto>> sendMessage(
            @Body SendChatMessageRequestDTO req
    );
}

File: app/src/main/java/com/example/product_sale_app/data/ChatApiService.kt
package com.example.product_sale_app.data

import retrofit2.http.*

interface ChatApiService {
    @GET("api/chat")
    suspend fun getMessages(
        @Query("pageIndex")  pageIndex: Int,
        @Query("pageSize")   pageSize: Int,
        @Query("chatBoxId")  chatBoxId: Int? = null,
        @Query("userId")     userId: Int?     = null
    ): BaseResponseModel<ChatMessageDto>

    @GET("api/chat/box/{chatBoxId}")
    suspend fun getBoxMessages(
        @Path("chatBoxId") boxId: Int,
        @Query("pageIndex") pageIndex: Int = 1,
        @Query("pageSize")  pageSize: Int  = 20
    ): BaseResponseModel<ChatMessageDto>

    @POST("api/chat")
    suspend fun sendMessage(
        @Body req: SendChatMessageRequestDTO
    ): BaseResponseModel<ChatMessageDto>
}
